### ISSUE TITLE
Prometheus & FireQOS Units Question

### ISSUE THREAD

**Issue by GitStoph** [firehol/netdata#1876]
Hi.

So I'm running FireQOS on my server, and classified several types of traffic. I'm attempting to use Prometheus/Grafana for historicals/graphing etc. I'm struggling a little with the units of measurement. Hoping someone could help.

I go to http://my.netdata.fqdn/api/v1/allmetrics?format=prometheus and locate the counters:
```
#netdata
# TYPE tc_eth0_21_ counter
tc_eth0_21_{instance="my_instance"} 198951955 1488383474010
#mysql
# TYPE tc_eth0_13_ counter
tc_eth0_13_{instance="my_instance"} 24101984648 1488383474010
```

So my Grafana expression for netdata ends up being:
`irate(tc_eth0_21_{instance="my.ip:19999",job="ND-MDMs"}[15s])`
For mysql I'm using:
`irate(tc_eth0_13_{instance="10.1.5.121:19999",job="ND-MDMs"}[15s])`

For axes I'm using bits/s. When I look at the Max/Avg/Total for Netdata that aligns with what i'm seeing in the live Netdata Quality of Service Monitor (81.9 kbps|47.6 kbps|26.5605 Mbps in 30min) . However, MySQL which is doing much more traffic, does NOT look correct (1.140 Mbps|291 kbps|162.630 Mbps in 30min). MySQL in the Netdata live QOS monitor is doing more around 425,000 kbps. I understand that the nature of the FireQOS collection can cause spikes that are likely double of what was realistic, but this is off by a lot. What am I missing? Prometheus is set to scrape every 5 seconds.
Thanks!

**Comment by ktsaou** [comment]
prometheus is getting the values "as-collected". This means these values are whatever the source reports. Are not normalized, interpolated, or in any way altered by netdata.

`tc` reports `bytes`, as a counter value (`incremental` in netdata lingo, i.e. a value that is constantly incrementing). netdata converts this to kilobits/s by multiplying with 8 and dividing by 1024.

> I understand that the nature of the FireQOS collection can cause spikes that are likely double of what was realistic

no, this is not right. FireQoS is very precise. Run `fireqos status world-in` on your terminal. It includes encapsulation overheads (if set), but it is very accurate. Compare it with the interface speed. You will be surprised how accurate it is.

**Comment by GitStoph** [comment]
> netdata converts this to kilobits/s by multiplying with 8 and dividing by 1024 and of course calculates the incremental difference to the previous measurement.
Thanks for this. That did the trick. I multiplied by 8 and just set the axes unit to bit/s and everything was looking great afterwards.

Slightly related, perhaps you want a new issue for this, but has anyone figured out how to get FireQOS to install on OSX? i pulled down the git, ran the autogen.sh, ran `sudo ./configure` and i get the following:
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... autotool/install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '0' is supported by ustar format... yes
checking whether GID '0' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether make sets $(MAKE)... (cached) yes
checking whether ln -s works... yes
checking whether LN_S has working -r option... no
checking for default FIREHOL_AUTOSAVE... no default available
checking for default FIREHOL_AUTOSAVE6... no default available
configure: error: docs not built, use '--disable-doc --disable-man' or --enable-maintainer-mode
Next, I tried `sudo ./configure --enable-maintainer-mode` and get the following:
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... autotool/install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '0' is supported by ustar format... yes
checking whether GID '0' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... yes
checking whether make sets $(MAKE)... (cached) yes
checking whether ln -s works... yes
checking whether LN_S has working -r option... no
checking for default FIREHOL_AUTOSAVE... no default available
checking for default FIREHOL_AUTOSAVE6... no default available
configure: ***************** MAINTAINER MODE *****************
checking for xmllint... /usr/bin/xmllint
checking for pandoc... /usr/local/bin/pandoc
checking whether /usr/local/bin/pandoc version is 1.12.2.1 or newer... yes (1.19.2.1)
checking for pandoc PDF output... yes
checking for pandoc HTML output... yes
checking for pandoc manpage output... yes
configure: ***************************************************
checking for bash... /bin/bash
checking whether /bin/bash version is 4.0 or newer... no (3.2)
configure: error: could not find required version of bash

This is further than I got last night after brew installing pandoc.. I'm still digging. I see MacPorts on github for bringing in shells, but I havent seen how to revert back to an older shell yet.. Any help is welcome!
